storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,28 +1,276 @@
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * @return the access policies of the queue
+     */
+    public Flux<SignedIdentifier> getAccessPolicy() {
+        return client.queues().getAccessPolicyWithRestResponseAsync(queueName, Context.NONE)
+            .flatMapMany(response -> Flux.fromIterable(response.value()));
+    }
+
+    /**
+     * Sets access policies of the queue
+     * @param permissions Access policies to set on the queue
+     * @return an empty response
+     */
+    public Mono<VoidResponse> setAccessPolicy(List<SignedIdentifier> permissions) {
+        return client.queues().setAccessPolicyWithRestResponseAsync(queueName, permissions, null,  null, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * Deletes all messages in the queue
+     * @return an empty response
+     */
+    public Mono<VoidResponse> clearMessages() {
+        return client.messages().clearWithRestResponseAsync(queueName, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * Adds a message to the queue
+     * @param messageText Message text
+     * @return the enqueued message information
+     */
+    public Mono<Response<EnqueuedMessage>> enqueueMessage(String messageText) {
+        return enqueueMessage(messageText, Duration.ofSeconds(0), Duration.ofDays(7));
+    }
+
+    /**
+     * Adds a message to the queue
+     * @param messageText Message text
+     * @param visibilityTimeout How long the message is invisible in the queue in seconds, default is 0 seconds
+     * @param timeToLive How long the message will stay in the queue in seconds, default is 7 days
+     * @return the enqueued message information
+     */
+    public Mono<Response<EnqueuedMessage>> enqueueMessage(String messageText, Duration visibilityTimeout, Duration timeToLive) {
+        Integer visibilityTimeoutInSeconds = (visibilityTimeout == null) ? null : (int) visibilityTimeout.getSeconds();
+        Integer timeToLiveInSeconds = (timeToLive == null) ? null : (int) timeToLive.getSeconds();
+        QueueMessage message = new QueueMessage().messageText(messageText);
+
+        return client.messages().enqueueWithRestResponseAsync(queueName, message, visibilityTimeoutInSeconds, timeToLiveInSeconds, null, null, Context.NONE)
+            .map(response -> mapResponse(response, response.value().get(0)));
+    }
+
+    /**
+     * Retrieves a message from the queue
+     * @return dequeued message information
+     */
+    public Flux<DequeuedMessage> dequeueMessages() {
+        return dequeueMessages(1, Duration.ofSeconds(30));
+    }
+
+    /**
+     * Retrieves a maximum number of messages from the queue
+     * @param maxMessages Maximum number of messages to get, must be in the range (0, 32], default is 1
+     * @return dequeued message information
+     */
+    public Flux<DequeuedMessage> dequeueMessages(Integer maxMessages) {
```

Comment:
Any particular reason why Integer is used in the API? If possible, can we use primitive types everywhere.

Reply:
maxMessages is an optional parameter, in the simplest overload I pass null through so the service can make the determination on the number of messages to dequeue.